backend/src/gamestate/player.rs

```diff
@@ -0,0 +1,77 @@
+use crate::cards::card::Card;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
+pub struct Player {
+    name: String,
+    pub is_author: bool,
+    cards: Vec<Card>,
+    position: Option<usize>,
+}
+
+impl Player {
+    pub fn new(name: String, is_author: bool) -> Player {
+        Player {
+            name,
+            is_author,
+            cards: vec![],
+            position: None,
+        }
+    }
+
+    /// Function returns Err if index is out of bounds
+    pub fn play_card_by_index(&mut self, index: usize) -> anyhow::Result<Card> {
+        if index >= self.cards.len() {
+            anyhow::bail!(
+                "Index {} out of bounds for card vec size of {}",
+                index,
+                self.cards.len()
+            )
+        }
+
+        Ok(self.cards.remove(index))
+    }
+
+    /// Function returns Err if card is not owned by the player.
+    pub fn play_card_by_eq(&mut self, card: Card) -> anyhow::Result<Card> {
+        let maybe_position = self.cards.iter().position(|c| c == &card);
+
+        match maybe_position {
+            None => anyhow::bail!(
+                "Card {:?} was not found in player {}'s hands.",
+                card,
+                self.name
+            ),
+            Some(position) => self.play_card_by_index(position),
+        }
+    }
+
+    pub fn give_card(&mut self, card: Card) {
+        self.cards.push(card)
+    }
+
+    pub fn is_finished(&self) -> bool {
+        self.position != None
+    }
+
+    /// Function panics if player.position == Option::None
+    pub fn get_position_unsafe(&self) -> usize {
+        self.position.unwrap()
+    }
+
+    pub fn set_position(&mut self, position: usize) {
+        self.position = Some(position)
+    }
+
+    pub fn get_card_count(&self) -> usize {
+        self.cards.len()
+    }
+
+    pub fn get_cards_clone(&self) -> Vec<Card> {
+        self.cards.clone()
+    }
+
+    pub fn get_name_clone(&self) -> String {
+        self.name.clone()
+    }
```

Comment:
Budeme někdy chtít jméno hráče, aniž bychom neklonovali? Jinými slovy, nestačí to pojmenovat `get_name()`, nebo, podle [naming conventions](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter), jenom `name()`?

To samé platí i pro ostatní funkce.

Reply:
Pokud budeme někdy chtít jméno hráče, aniž bychom ho clonovali, pravděpodobně nás plácne přes prsty borrow-checker se zprávou obsahující "cannot move" či "is behind shared reference". To samé platí i pro ostatní funkce: udělal jsem je s cílem nemít public atributy struktury Player, a k tomuto jsem došel.

Reply:
Actually, didn't know that naming convention... cool.

Reply:
> Pokud budeme někdy chtít jméno hráče, aniž bychom ho clonovali, pravděpodobně nás plácne přes prsty borrow-checker se zprávou obsahující "cannot move" či "is behind shared reference". To samé platí i pro ostatní funkce zmíněné v review: udělal jsem je s cílem nemít public atributy struktury Player, jak bylo zmíněno v #16, a k tomuto stavu jsem došel.

Proto jsem psal "aniž bychom ho **ne**klonovali", šlo mi o to se zbavit `_clone` :)

Reply:
Ah, už chápu. Will rename.